core/src/main/java/com/linecorp/armeria/common/http/HttpParameters.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.http;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Map;
+
+import io.netty.handler.codec.Headers;
+
+/**
+ * HTTP parameters map.
+ */
+public interface HttpParameters extends Headers<String, String, HttpParameters> {
+
+    /**
+     * An immutable empty HTTP parameters map.
+     */
+    HttpParameters EMPTY_PARAMETERS = new DefaultHttpParameters().asImmutable();
+
+    /**
+     * Returns new HTTP parameters map.
+     */
+    static HttpParameters of() {
+        return new DefaultHttpParameters();
+    }
+
+    /**
+     * Returns new HTTP parameters map with {@code Map<String, List<String>>}.
+     */
+    static HttpParameters of(Map<String, List<String>> parameters) {
```

Comment:
- Could be `Map<String, ? extends Iterable<String>>`?
- I guess `copyOf` is better given that `ImmutableList` uses `copyOf()` for a similar case.